bound/bound_test.go

```diff
@@ -126,3 +126,52 @@ func TestUnion(t *testing.T) {
 func same(a, b Bound) bool {
 	return floats.Same([]float64{a.Min, a.Max}, []float64{b.Min, b.Max})
 }
+
+var jaccardTests = []struct {
+	bounds []Bound
+	want   float64
+}{
+	{
+		bounds: []Bound{{Min: 0, Max: 5}},
+		want:   1.0,
+	},
+	{
+		bounds: []Bound{{Min: 0, Max: 5}, {Min: 3, Max: 4}},
+		want:   0.2,
+	},
+	{
+		bounds: []Bound{{Min: 0, Max: 5}, {Min: -1, Max: 1}},
+		want:   0.16666666666666666,
+	},
+	{
+		bounds: []Bound{{Min: -1, Max: 1}, {Min: 1, Max: 2}},
+		want:   0.0,
+	},
+	{
+		bounds: []Bound{{Min: -1, Max: 1}, {Min: 1.1, Max: 2}},
+		want:   math.NaN(),
+	},
+	{
```

Comment:
I need to rethink the approach. I would expect this to return 0. Maybe, rather than what I suggested, get the lengths of each of the union and intersection and convert to 0 if NaN, then do the Jaccard calculation.

Reply:
This still doesn't seem right. The intersection of this is empty, but the union is [0, 3]. So the index should be 0.

Reply:
@kortschak 
Current intersection implementation returns `{NaN NaN}`for `[{0.5 2.5} {0 1} {2 3}]`.
According to https://github.com/gonum/gonum/pull/969#issuecomment-485382802
NaN/3.0 == NaN.
Should we deal `{NaN NaN}` as empty bound?

Reply:
Yes, `{NaN NaN}` should be treated as empty. This is the problems we have, there is invalid empty and validly empty and we don't distinguish between them.